.github/workflows/main.yml

```diff
@@ -0,0 +1,435 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: NetBeans
+
+on:
+  push:
+  pull_request:
+
+jobs:
+    
+# primary build job, most other jobs use the artifact produced here
+  base-build:
+    name: Build Clusters on JDK 11
+    runs-on: ubuntu-latest
+    env:
+      ANT_OPTS: -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json
+    steps:
+        
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'zulu'
+
+      - name: Caching dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.hgexternalcache
+          key: ${{ runner.os }}-${{ hashFiles('*/external/binaries-list', '*/*/external/binaries-list') }}
+          restore-keys: ${{ runner.os }}-
+          
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Build NetBeans
+        run: ant -Dcluster.config=release build-nozip
+
+      - name: Archive Build
+        run: tar -czf /tmp/build.tar.gz --exclude ".git" .
+
+      - name: Upload Build
+        uses: actions/upload-artifact@v2
+        with:
+          name: build
+          path: /tmp/build.tar.gz
+          retention-days: 1
```

Comment:
@neilcsmith-net do you think it is a problem from the apache resources perspective that a build would create a 1.5 GB upload ([accessible from bottom of this page](https://github.com/apache/netbeans/actions/runs/1989525198))? 1 day retention is already the minimal setting.

Reply:
@matthiasblaesing @neilcsmith-net if I take a look at https://github.com/pricing

I see "Free for public repositories" under the Package storage item. Does this mean we shouldn't have to worry here?

I could try to add a cleanup job which removes the upload again. (for example a variant of: https://github.com/actions/upload-artifact/issues/5#issuecomment-876499762 but only for the current workflow)

Or should we give it a try first?

Reply:
@mbien I'd run it past infra@.  Not sure of the answer.  Pricing may not be relevant info - ASF is sponsored by GitHub (with enterprise org features IIRC) but what that means for reasonable resource usage in practice I don't know for sure.

Reply:
@neilcsmith-net excellent, meanwhile I am going to try to add a cleanup job: [9e719ee](https://github.com/apache/netbeans/pull/3785/commits/9e719ee57d8b1468b609cd264f4880901ec9e7d1)

Reply:
```
Run ./.github/actions/delete-artifact/
Successfully deleted artifact "build".
```
it works :)

Reply:
@neilcsmith-net ok to merge or should we wait for infra reply? I am no longer concerned about the storage issue due to the cleanup job which was much easier to implement than anticipated thanks to the third party gh action (deployed via the submodule trick).

Reply:
@mbien I reckon it's reasonable as is. Let's merge and deal with any fallout later! :smile: